flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/inmemory/InMemoryStateChangelogStorage.java

```diff
@@ -19,14 +19,14 @@
 
 import org.apache.flink.runtime.state.KeyGroupRange;
 import org.apache.flink.runtime.state.changelog.StateChangelogHandleReader;
-import org.apache.flink.runtime.state.changelog.StateChangelogWriterFactory;
+import org.apache.flink.runtime.state.changelog.StateChangelogStorage;
 import org.apache.flink.util.CloseableIterator;
 
 import java.io.Serializable;
 
-/** An in-memory (non-production) implementation of {@link StateChangelogWriterFactory}. */
-public class InMemoryStateChangelogWriterFactory
-        implements StateChangelogWriterFactory<InMemoryStateChangelogHandle>, Serializable {
+/** An in-memory (non-production) implementation of {@link StateChangelogStorage}. */
+public class InMemoryStateChangelogStorage
+        implements StateChangelogStorage<InMemoryStateChangelogHandle>, Serializable {
```

Comment:
Neither of them actually should be Serializable.
It was needed because instantiation was made in `StateBackendLoader`, but that was revisited (see FLINK-21804).
I'm going to remove `implements Serializable` from both.